lib/core/src/chain_swap.rs

```diff
@@ -465,18 +478,9 @@ impl ChainSwapStateHandler {
                         warn!("Chain Swap {id} is in an unrecoverable state: {swap_state:?}");
                         match swap.user_lockup_tx_id.clone() {
                             Some(_) => {
-                                warn!("Chain Swap {id} user lockup tx has been broadcast. Attempting refund.");
-                                let refund_tx_id = self.refund_outgoing_swap(swap).await?;
-                                info!("Broadcast refund tx for Chain Swap {id}. Tx id: {refund_tx_id}");
-                                self.update_swap_info(
-                                    id,
-                                    RefundPending,
-                                    None,
-                                    None,
-                                    None,
-                                    Some(&refund_tx_id),
-                                )
-                                .await?;
+                                warn!("Chain Swap {id} user lockup tx has been broadcast.");
```

Comment:
I think we should try here cooperative refund which should work for 99% only if fails signal to the non cooperative refund. Also cooperative refund should happen immediately for example when trying to pay a not online peer, the swapper will fail and then we need to refund immediately. Let's also make sure we test that as part of this PR.
Also why use lock and im memory slice? Isn't it better to just mark these in the db as failed to refund coopeartively and fetch them using a db query when needed?